src/sysen2/mldev.108

```diff
@@ -112,77 +111,67 @@ GO:	MOVE P,[-LPDLL-1,,PDL-1]
 	ldb c,[.bp (770000),a]
 	cain c,'X
 	 jrst [	move b,a
-		lsh b,6              ;get rid of X
-		pushj p,isitsp       ; see if we now have a known ITS name
-		 jrst nox            ;  nope, skip handling as experimental case
-		move c,dskdev        ; assume DSK device
-		movem c,jbcdev       ; store as remote device
-		move a,b	         ; get host in A for gothst
-		jrst gothst ]        ; and go process request for this host
-
-nox:                         ; jump here if hostname minus first "x" is not a known ITS
-                             ; check to see if format is hhDIR
+		lsh b,6		;get rid of X
+		pushj p,isitsp	; see if we now have a known ITS name
+		 jrst nox	;  nope, skip handling as experimental case
+		move c,dskdev	; assume DSK device
+		movem c,jbcdev	; store as remote device
+		move a,b	; get host in A for gothst
+		jrst gothst ]	; and go process request for this host
+
+nox:				; jump here if hostname minus first "x" is not a known ITS
+				; check to see if format is hhDIR
 	move c,a
-	lsh c,12.                ; move past hostname
-	camn c,dirdev            ; is what remains the DIR device?
-	  jrst dirchk            ;  yes, continue checking hhDIR case
-
-                             ; is it hhhdir or hhh
-	hrlz c,a
-	jumpe c, host3		     ; jump if no device
-	  jrst hst2ck            ;  have a device, nope, check for hhddd case
-	movem c,jbcdev           ; we have a device; save it as remote device
-	jrst chkits              ; see if host part (hhh) is a known ITS machine
-
-hst2ck:	                     ; check for hhddd case
-	ldb c,[000600,,a]        ; can only be hhddd if last sixbit character is null
-	cain c,0                 ; is last character null?
-	  jrst hhddd1            ;  yes, handle hhddd case
+	lsh c,12.		; move past hostname
+	camn c,dirdev		; is what remains the DIR device?
+	  jrst dirchk		;  yes, continue checking hhDIR case
+
+				; is it hhhdir or hhh
+	hrlz c,a		; get device in c
+	jumpe c, host3		; jump if no device
+	jrst hst2ck		;  have a device, nope, check for hhddd case
+
+hst2ck:				; check for hhddd case
+	ldb c,[000600,,a]	; can only be hhddd if last sixbit character is null
+	jumpn c,hhddd1		; handle hhddd case if last character is NUL
```

Comment:
Did I say `jumpn` here?  Sorry if so.  It should be `jumpe` right?

Reply:
I don’t recall what you said, but I tested all use cases and they worked. I think. I’ll check again and look at that code.

Reply:
Yes, should be a jumpe.  I wonder why I was able to test the EXSHO case, and have it work?  I'll step through that again and see why the JUMPN seemed to work just fine.  May be a place for more optimization.  

There is some extra case handling that could probably be removed now -- the old code handled the hhhDIR case specially, but now the more general hhhDDD handling shouid cover that case.  Same with the hhDIR versus hhDDD cases.  These were separate before and I left them separate.  May optimize this later.

Reply:
I tested the code again and the HHDDD case (e.g. EXSHO) case failed due to the JUMPN (which should have been a JUMPE).  I realized that I had tested this case out using a slightly older version of the ATSIGN MLDEV binary.  I'm running through all test cases again.  I also got rid of the hst2ck label, since it wasn't used.